### PR TITLE
Preparing all the small things for release

### PR DESCRIPTION
**Pull request by gnunicorn**
This PR contains all the works to make our crates ready for publishing to crates.io:

- [x] clean up `Cargo.toml` required Metadata, (e.g. `license`, `repository`)
- [x] set readme files for new crates
- [x] fix test-macros error
- [x] bump-versions where necessary
- [x] ensure everything builds (thanks to `cargo unleash`)

### CODE REVIEW

**Comment by gnunicorn** [comment]
Found an error with the test-macros:

```
cargo check -p matrix-sdk-test-macros
    Blocking waiting for file lock on package cache
    Blocking waiting for file lock on package cache
    Blocking waiting for file lock on package cache
    Checking matrix-sdk-test-macros v0.2.0 (/home/ben/Dev/matrix/matrix-rust-sdk/crates/matrix-sdk-test-macros)
error[E0369]: binary operation `==` cannot be applied to type `ReturnType`
  --> crates/matrix-sdk-test-macros/src/lib.rs:11:23
   |
11 |     if fun.sig.output == syn::ReturnType::Default {
   |        -------------- ^^ ------------------------ ReturnType
   |        |
   |        ReturnType

For more information about this error, try `rustc --explain E0369`.
error: could not compile `matrix-sdk-test-macros` due to previous error
```

**Comment by jplatte** [comment]
Right. Needs the `extra-traits` feature on syn, or using `if let` instead of `==`. In regular (workspace) builds `extra-traits` gets enabled through Ruma, I think.

File: crates/matrix-sdk-sled/Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,12 @@
 name = "matrix-sdk-sled"
 version = "0.1.0"
 edition = "2021"
+authors = ["Damir Jelić <poljar@termina.org.uk>"]
+repository = "https://github.com/matrix-org/matrix-rust-sdk"
+description = "Skled Storage backend for matrix-sdk for native environments"
```
Comment (jplatte) [inline crates/matrix-sdk-sled/Cargo.toml:7]:
```suggestion
description = "Sled Storage backend for matrix-sdk for native environments"
```

File: crates/matrix-sdk-sled/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# matrix-sdk-sled
+
+This crate implements a storage backend using [sled][sled] for native and mobile environments using the matrix-sdk-base primitives. When using **matrix-sdk** this is included by default.
+
+_Note_: the future of [sled][sled] is unclear. While it is currently the default for mobile and native environments for matrix-rust-sdk, [the team is actively looking at replacing it with a different storage backend](https://github.com/matrix-org/matrix-rust-sdk/issues/294).
```
Comment (jplatte) [inline crates/matrix-sdk-sled/README.md:5]:
```suggestion
_Note_: the future of [sled][sled] is unclear. While it is currently the default for mobile and native environments for matrix-rust-sdk, [we are actively looking at replacing it with a different storage backend](https://github.com/matrix-org/matrix-rust-sdk/issues/294).
```

File: crates/matrix-sdk-store-encryption/Cargo.toml [hunk 3]
```diff
@@ -2,6 +2,7 @@
 name = "matrix-sdk-store-encryption"
 version = "0.1.0"
 edition = "2021"
+authors = ["Damir Jelić <poljar@termina.org.uk>"]
```
Comment (jplatte) [inline crates/matrix-sdk-store-encryption/Cargo.toml:5]: I don't think we should add authors fields. It's not required for publishing and git history (and possibly copyright headers) is much better at tracking authorship.

Comment (gnunicorn) [reply crates/matrix-sdk-store-encryption/Cargo.toml:5]:
```suggestion
```

File: crates/matrix-sdk-crypto-ffi/Cargo.toml [hunk 4]
```diff
@@ -4,6 +4,9 @@ version = "0.1.0"
 authors = ["Damir Jelić <poljar@termina.org.uk>"]
 edition = "2018"
 rust-version = "1.60"
+description = "Uniffi based bindings for the Rust SDK crypto crate"
+repository = "https://github.com/matrix-org/matrix-rust-sdk"
+license = "Apache-2.0"
```
Comment (gnunicorn) [inline crates/matrix-sdk-crypto-ffi/Cargo.toml:9]:
```suggestion
license = "Apache-2.0"
publish = false
```

**Review by jplatte — APPROVED** [review]
One more missing period, otherwise LGTM 👍🏼

**Pull request merged**